tests/Engagement_unit_test.py

```diff
@@ -0,0 +1,141 @@
+        Select(driver.find_element_by_id("id_scan_type")).select_by_visible_text("Nmap Scan")
+        driver.find_element_by_id('id_file').send_keys(os.getcwd().replace('/tests', '') + "/dojo/unittests/scans/nmap_sample/nmap_multiple_port.xml")
+        driver.find_element_by_css_selector("input.btn.btn-primary").click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Nmap Scan processed, a total of 13 findings were processed', EngagementTXT))
+
+    def test_close_new_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_class_name("pull-left").click()
+        driver.find_element_by_link_text("View Engagements").click()
+        driver.find_element_by_link_text("edited test engagement").click()
+        driver.find_element_by_id("dropdownMenu1").click()
+        driver.find_element_by_link_text("Close Engagement").click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement closed successfully.', EngagementTXT))
+
+    def test_delete_new_closed_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_class_name("pull-left").click()
+        driver.find_element_by_link_text('View Engagements').click()
+        driver.find_element_by_link_text("edited test engagement").click()
+        driver.find_element_by_id("dropdownMenu1").click()
+        driver.find_element_by_link_text('Delete Engagement').click()
+        driver.find_element_by_name('delete_name').click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement and relationships removed.', EngagementTXT))
+
+    def test_new_ci_cd_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_link_text('Engagement Test').click()
+        driver.find_element_by_xpath("//a[@class='dropdown-toggle active']//span[@class='hidden-xs']").click()
+        driver.find_element_by_link_text('Add New CI/CD Engagement').click()
+        driver.find_element_by_id("id_name").send_keys("test new ci/cd engagement")
+        driver.find_element_by_id("id_name").send_keys("\ttest new ci/cd engagement")
+        driver.find_element_by_id('id_deduplication_on_engagement').get_attribute('checked')
+        driver.find_element_by_css_selector("input[value='Done']").click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement added successfully.', EngagementTXT))
+
+    def tearDown(self):
+        self.driver.quit()
+        self.assertEqual([], self.verificationErrors)
+
+
+def suite():
+    suite = unittest.TestSuite()
+    suite.addTest(EngagementTest('create_new_product_to_test_engagement'))
```

Comment:
Be sure to clean up used resources. Deleting a product also deletes all engagements/findings/.etc.

Reply:
yes, that's why I created a new product and test all engagement separately.